lib/packages/shared-utils/legacy-event-type.ts

```diff
@@ -0,0 +1,56 @@
+import { Action } from "shared-types";
+
+// Resolve the action type based on the GSI1pk
+export const getLegacyEventType = (GSI1pk: string) => {
+  if (!GSI1pk) {
+    return undefined;
+  }
+  const submitType = GSI1pk?.split("OneMAC#submit")?.[1] || "";
+
+  let event;
+
+  switch (submitType) {
+    case "chipspa":
+    case "medicaidspa":
+    case "waiveramendment":
+    case "waiverappk":
+    case "waiverextension":
+    case "waiverextensionb":
+    case "waiverextensionc":
+    case "waivernew":
+    case "waiverrenewal":
+      event = "new-legacy-submission";
```

Comment:
Why not just return the appropriate string as opposed to re-assigning `event`? I think it'd be easier to read